src/data/moves/move-attrs/add-substitute-attr.ts

```diff
@@ -42,9 +47,15 @@ export class AddSubstituteAttr extends MoveEffectAttr {
   }
 
   override getCondition(): MoveConditionFunc {
+    /**
+     * Only works if
+     * - The user does not have a substitute out already
+     * - The user has enough HP
+     * - THe user has more than 1 max hp (wonder guard users)
+     */
     return (user, _target, _move) =>
       !user.getTag(BattlerTagType.SUBSTITUTE)
-      && user.hp > Math.floor(user.getMaxHp() * this.hpCost)
+      && user.hp > (this.forShedTail ? Math.ceil(user.getMaxHp() * 0.5) : Math.floor(user.getMaxHp() * 0.25))
```

Comment:
Apply DRY (Do not repeat yourself) principal here. 
The code
```ts
this.forShedTail ? Math.ceil(user.getMaxHp() * 0.5) : Math.floor(user.getMaxHp() * 0.25)
```

is used twice. Should probably create a private helper method returning the hpCost